pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -252,6 +295,31 @@ func (r *ReconcileOneAgent) reconcileRollout(logger logr.Logger, instance *dynat
 	return updateCR, nil
 }
 
+func (r *ReconcileOneAgent) determineOneAgentPhase(instance *dynatracev1alpha1.OneAgent) (bool, error) {
+	// Check if this DaemonSet already exists
+	var phaseChanged bool
+	dsActual := &appsv1.DaemonSet{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, dsActual)
+	if err != nil && !errors.IsNotFound(err) {
+		phaseChanged = instance.Status.Phase != dynatracev1alpha1.Error
+		instance.Status.Phase = dynatracev1alpha1.Error
+		return phaseChanged, err
+	}
+
+	// Case: Daemonset exists
+	if !errors.IsNotFound(err) {
+		if dsActual.Status.NumberReady == dsActual.Status.CurrentNumberScheduled {
```

Comment:
There are some other scenarios, such as, what if `CurrentNumberScheduled` equals 0 (i.e., can't deploy on any node) - but we can take them as we go on.